astropy/io/fits/tests/test_table.py

```diff
@@ -3349,6 +3349,40 @@ def test_heterogeneous_VLA_tables(self):
         ):
             fits.BinTableHDU.from_columns([c1])
 
+    def test_vla_slice(self):
+        filename = self.data("variable_length_table.fits")
+
+        # slicing the VLA array
+        with fits.open(filename) as hdul:
+            hdu = hdul[1]
+            arr = hdu.data["var"][:1]
+            assert arr.tolist() == [[45, 56]]
+            assert arr[0].dtype == np.int16
+
+        # slicing .data
+        with fits.open(filename) as hdul:
+            hdu = hdul[1]
+            arr = hdu.data[:1]["var"]
+            assert arr.tolist() == [[45, 56]]
+            assert arr[0].dtype == np.int16
+
+    def test_vla_slice2(self):
```

Comment:
couldn't these test be merged into a single parametrized function ?

Reply:
This would make things harder to read and understand. If there were more than 2 maybe but here simple is better imo.

Reply:
fine by me